src/libs/actions/Report.js

```diff
@@ -544,16 +546,23 @@ function openReport(reportID, participantList = [], newReportObject = {}) {
         onyxData.optimisticData[0].onyxMethod = CONST.ONYX.METHOD.SET;
 
         // Also create a report action so that the page isn't endlessly loading
+        const optimisticReportAction = ReportUtils.buildOptimisticCreatedReportAction(newReportObject.ownerEmail);
```

Comment:
We should update `buildOptimisticCreatedReportAction` to add the `reportActionID` to the reportAction, rather than generating it in here and assigning it as a property after

Reply:
Is this([13224](https://github.com/Expensify/App/issues/13224#issuecomment-1338323177)) something similar are you referring to?

Reply:
Can be set here https://github.com/Expensify/App/blob/cb9a54138bafac9d39598e492669cd01a9b31af5/src/libs/ReportUtils.js#L869-L875 as a property and then we won't need `optimisticReportActionID`

Reply:
Done, but we still need `optimisticReportActionID` so that it always exists in the scope outside of this `if`.

Reply:
Can we make `buildOptimisticCreatedReportAction` return just the action instead of `0: { actionName, ... }`? We will need next to update the code that sets the action in Onyx this way. Anything wrong with this approach?

```
optimisticData[1].onyxMethod = CONST.ONYX.METHOD.SET;
optimisticData[1].value = {
    0: {
        ...optimisticCreatedAction,
    },
    ...optimisticData[1].value,
};
```